src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -603,23 +603,30 @@ async def ConnectBLE(self, discriminator: int, setupPinCode: int, nodeid: int, i
 
         async with self._commissioning_context as ctx:
             self._enablePairingCompleteCallback(True)
-            res = await self._ChipStack.CallAsync(
-                lambda: self._dmLib.pychip_DeviceController_ConnectBLE(
-                    self.devCtrl, discriminator, isShortDiscriminator, setupPinCode, nodeid)
-            )
-            res.raise_on_error()
+            try:
+                await self._ChipStack.CallAsync(
+                    lambda: self._dmLib.pychip_DeviceController_ConnectBLE(
+                        self.devCtrl, discriminator, isShortDiscriminator, setupPinCode, nodeid)
+                )
+            except:
```

Comment:
Can you narrow the scope of this catch (in particular not to catch `KeyboardInterrupt`)?
In other places as well.

Reply:
It is not catching any exception in that sense, as we re-raise immediately.

If we would narrow this down to `KeyboardInterrupt`, We'd end up with a `RuntimeError` from the CommissionContext.

Maybe it would be better to just get rid of the RuntimeError there actually :thinking:

Reply:
Yeah thinking a bit more about it, I think this is the better way:

```patch
     async def __aexit__(self, exc_type, exc_value, traceback):
         if not self._future.done():
-            raise RuntimeError("CallbackContext future not completed")
+            # In case the initial call (which sets up for the callback) fails,
+            # the future will never be used actually. So just cancel it here
+            # for completeness, in case somebody is expecting it to be completed.
+            self._future.cancel()
         self._future = None
         self._lock.release()
```

Then we don't need this `try .. except` everywhere.

Reply:
Changed, also updated the PR description.